### PR TITLE
Migrate to using HttpClientsConfig.

### PR DESCRIPTION
**Pull request by mpawliszyn**
This opens the door to replacing the connectorExtraData with just a url field I believe.

### CODE REVIEW

File: service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt [hunk 1]
```diff
@@ -2,23 +2,25 @@ package app.cash.backfila.client
...
 import retrofit2.adapter.guava.GuavaCallAdapterFactory
 import retrofit2.converter.wire.WireConverterFactory
 
 @Singleton
 class EnvoyClientServiceClientProvider @Inject constructor(
-  @Named(ENVOY) private val defaultHttpClientEndpointConfig: HttpClientEndpointConfig,
+  private val httpClientsConfig: HttpClientsConfig,
   private val httpClientFactory: HttpClientFactory,
-  private val httpClientConfigUrlProvider: HttpClientConfigUrlProvider,
+  private val envoyClientEndpointProvider: EnvoyClientEndpointProvider,
```
Comment (mpawliszyn) [inline service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt:23]: @shellderp what do you think of replacing connectorExtraData with just a url field?

Comment (shellderp) [reply service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt:23]: sounds trivial for http

```
data class HttpConnectorData(val url: String)

data class EnvoyConnectorData(val clusterType: String)
```

for envoy would `clusterType` be the "url"?

Comment (shellderp) [reply service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt:23]: seems reasonable, ultimately it's just a callback url

File: service/src/main/kotlin/app/cash/backfila/client/BackfilaClientServiceClientProvider.kt [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
 
-val DEFAULT_HTTP_CLIENT_ENDPOINT_CONFIG = HttpClientEndpointConfig(
-    // Allow RunBatch requests to take a long time.
-    readTimeout = Duration.ofSeconds(30L),
-    // Use a large number, we limit parallel calls ourselves.
-    maxRequests = 1000,
-    maxRequestsPerHost = 1000
+val DEFAULT_HTTP_CLIENTS_CONFIG = HttpClientsConfig(
+  hostConfigs = linkedMapOf(
+    ".*" to HttpClientConfig(
```
Comment (shellderp) [inline service/src/main/kotlin/app/cash/backfila/client/BackfilaClientServiceClientProvider.kt:13]: huh, is regex in the key a new feature?

Comment (mpawliszyn) [reply service/src/main/kotlin/app/cash/backfila/client/BackfilaClientServiceClientProvider.kt:13]: The idea is that backfila is no longer in the middle. Its how you define what urls get what config.

File: service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt (hunk 1 above)
Comment (mpawliszyn) [reply service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt:23]: Well I was thinking of just replacing the data type completely. Although now I'm thinking that is just something to do when we run into the next client type?

Unfortunately the retrofit clients are still a bit different between the envoy and http clients. Unless you think that should be specified too.

File: service/src/main/kotlin/app/cash/backfila/client/BackfilaClientServiceClientProvider.kt (hunk 2 above)
Comment (bezmax) [reply service/src/main/kotlin/app/cash/backfila/client/BackfilaClientServiceClientProvider.kt:13]: Yeah, we tried to make routing of requests abstracted away. For this to work, I changed the HttpClientsConfig to be URL based. This way, Skim can define default routing settings (like `*.gns.square` always going through the abstract socket), and clients can just provide an URL and delegate the correct routing of this URL to Misk/Skim/OS.

File: service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt (hunk 1 above)
Comment (bezmax) [reply service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt:23]: @mpawliszyn Could you clarify why retrofit clients are different? From what I've seen, to build a retrofit client you still go to HttpClientsConfig to ask it for config, and they you build the client from that. Therefore, same approach still applies - as long as you know the URL you can build the client. Or am I missing something?

Comment (shellderp) [reply service/src/main/kotlin/app/cash/backfila/client/EnvoyClientServiceClientProvider.kt:23]: yea I meant replace data structure with url. exactly what does the url look like for the envoy client though? I suppose that's not my problem :p

**Pull request merged**